consensus/src/chained_bft/consensus_types/block.rs

```diff
@@ -225,6 +225,14 @@ where
         // A block must carry a QC to its parent.
         checked_precondition_eq!(quorum_cert.certified_block_id(), parent_block.id());
         checked_precondition!(round > parent_block.round());
+
+        // This precondition guards the addition overflow caused by passing
+        // "parent_block.height() + 1" to "new_internal". Since the
+        // "height" and "round" values are always same, the check
```

Comment:
> "height" and "round" values are always same

That's not correct! Block timeouts and result in a new-view, which result in a new round for the same block proposal. The height is the distance induced by transitive parenthood to the genesis block.  The round is an order on block proposals.